nova/virt/vmwareapi/vmops.py

```diff
@@ -2014,10 +2007,47 @@ def finish_migration(self, context, migration, instance, disk_info,
         """Completes a resize, turning on the migrated instance."""
         vm_ref = vm_util.get_vm_ref(self._session, instance)
 
-        # 5. Update ephemerals if necessary
+        flavor = instance.flavor
+        boot_from_volume = compute_utils.is_volume_backed_instance(context,
+                                                                   instance)
+        reattach_volumes = False
+        # 2. Relocate the VM if necessary
```

Comment:
Why do we do the migration in the `finish_migration()` step and not the `migrate_disk_and_power_off()` step where `libvirt` does it? To me it seems like this is not the correct function for implementing the migration itself.

Reply:
I assume it's because of the `network_info`. Maybe, we could instead set some predefined network during the migration and then update the configuration as intended in `finish_migration()`. The VM is powered off anyways, so there should be no harm.

Reply:
One of the things `_relocate_vm()` does it that it also selects a datastore (typically used to move ephemeral disks).
Selecting a datastore is done by also using the `datastore_regex` config, which is a property configured locally in the `dest_compute` service. Since `migrate_disk_and_power_off()` is invoked in `source_compute`, at that point I do not see a way for us accessing the `dest_compute` `datastore_regex` config.

Reply:
Good point. But we can always call the destination host to ask for a valid datastore or even hand over the task of migrating the disk.